lib/IRGen/GenDistributed.cpp

```diff
@@ -53,3 +61,294 @@ llvm::Value *irgen::emitDistributedActorInitializeRemote(
+
+    // 2. Cast the pointer to the type of the element.
+    Address eltPtr = IGF.Builder.CreateBitCast(
+        Address(currentOffset, IGM.getPointerAlignment()),
+        IGM.getStoragePointerType(paramTy));
+
+    // 3. Adjust typed pointer to the alignement of the type.
+    auto alignedOffset = typeInfo.roundUpToTypeAlignment(IGF, eltPtr, paramTy);
+
+    // 4. Load argument value from the element pointer.
+    auto *argValue = IGF.Builder.CreateLoad(alignedOffset, "argval");
+
+    // 5. Create an exploded version of the type to pass as an
+    //    argument to distributed method.
+
+    if (paramTy.isObject()) {
+      auto &nativeSchema = typeInfo.nativeParameterValueSchema(IGM);
+      auto expandedTy = nativeSchema.getExpandedType(IGM);
+
+      if (nativeSchema.requiresIndirect()) {
+        llvm_unreachable("indirect parameters are not supported");
+      }
+
+      Explosion nonNativeParam;
+      nonNativeParam.add(argValue);
+
+      // Convert SIL type into a native representation.
+      auto nativeParam = nativeSchema.mapIntoNative(
+          IGM, IGF, nonNativeParam, paramTy, /*isOutlined=*/false);
+
+      // If expanded type is a struct we need to explode it to match
+      // expected argument schema.
+      if (auto *ST = dyn_cast<llvm::StructType>(expandedTy)) {
+        IGF.emitAllExtractValues(nativeParam.claimNext(), ST, arguments);
+      } else {
+        arguments.add(nativeParam.claimNext());
+      }
+    } else {
+      arguments.add(argValue);
+    }
+
+    // 6. Move the offset to the beginning of the next element.
+    {
+      llvm::Value *typeSize = typeInfo.getSize(IGF, paramTy);
+
+      llvm::Value *addr = alignedOffset.getAddress();
+      addr = IGF.Builder.CreatePtrToInt(addr, IGM.IntPtrTy);
+      llvm::Value *nextOffset = IGF.Builder.CreateIntToPtr(
+          IGF.Builder.CreateAdd(addr, typeSize), IGM.Int8PtrTy);
+
+      IGF.Builder.CreateStore(nextOffset, offset);
+    }
+  }
+
+  IGF.Builder.CreateLifetimeEnd(offset, IGM.getPointerSize());
+}
+
+void DistributedAccessor::emit() {
+  auto methodTy = Method->getLoweredFunctionType();
+  SILFunctionConventions conv(methodTy, IGF.getSILModule());
+  TypeExpansionContext expansionContext = IGM.getMaximalTypeExpansionContext();
+
+  auto params = IGF.collectParameters();
+
+  auto directResultTy = conv.getSILResultType(expansionContext);
+  const auto &directResultTI = IGM.getTypeInfo(directResultTy);
+  auto &resultSchema = directResultTI.nativeReturnValueSchema(IGM);
+  llvm::Value *indirectResultSlot = nullptr;
+
+  if (resultSchema.requiresIndirect())
+    indirectResultSlot = params.claimNext();
+
+  Explosion arguments;
+  // Claim indirect results first, they are going to be passed
+  // through to the distributed method.
+  params.transferInto(arguments, conv.getNumIndirectSILResults());
+
+  // FIXME: Once we remove async task and executor this should be one not three.
```

Comment:
Async task and executor are long gone... is this a stale FIXME that was copied?

Reply:
Yeah, I copied that from `GenThunk.cpp`. I'll remove it from here and there too.

Reply:
Fixed!